pkg/image/admission/imagepolicy/api/v1/types.go

```diff
@@ -0,0 +1,129 @@
+	ImageSourceRule `json:",inline"`
+
+	Add []ConsumeResourceEffect `json:"add"`
+}
+
+type ConsumeResourceEffect struct {
+	// Name is the name of a quantity to set on each matching container.
+	// May not be specified if NameFromImageAnnotation or NameFromDockerImageLabel is set.
+	Name string `json:"name"`
+	// NameFromImageAnnotation is the name of an image annotation to use to set the name of the resource
+	// quantity on each matching container.
+	// May not be specified if Name or NameFromDockerImageLabel is set.
+	NameFromImageAnnotation string `json:"nameFromImageAnnotation"`
+	// NameFromDockerImageLabel is the name of a docker image label to use to set the name of the resource
+	// quantity on each matching container.
+	// May not be specified if Name or NameFromImageAnnotation is set.
+	NameFromDockerImageLabel string `json:"nameFromDockerImageLabel"`
+
+	// Quantity is the amount of quantity to set
+	Quantity string `json:"quantity"`
+	// QuantityFromImageAnnotation is the key on an image annotation to use for the value of this resource.
+	// If this value is specified and no annotation is found, the value of Quantity is used instead.
+	QuantityFromImageAnnotation string `json:"quantityFromImageAnnotation"`
+	// QuantityFromDockerImageLabel is the key on a docker image label to use for the value of this resource.
+	// If this value is specified and no annotation is found, the value of Quantity is used instead.
+	QuantityFromDockerImageLabel string `json:"quantityFromDockerImageLabel"`
+}
+
+type ImageExecutionPolicyRule struct {
+	ImageSourceRule `json:",inline"`
+
+	// Deny indicates this rule is inverted and a match results in a rejection
+	Deny bool `json:"deny"`
+
+	// Resolve indicates that images referenced by this resource must be resolved
+	Resolve bool `json:"resolve"`
+}
+
+// ImageSourceRule defines the conditions for matching a particular image source. The conditions below
+// are all required (logical AND)
+type ImageSourceRule struct {
+	// Name is the name of this policy rule for reference. It must be unique across all rules.
+	Name string `json:"name"`
+	// OnResources determines which resources this applies to. Defaults to 'pods' for RuntimeSourceRules.
+	OnResources []string `json:"onResources"`
+	// IgnoreNamespaceOverride prevents this rule from being overriden when the
+	// `alpha.image.policy.openshift.io/ignore-rules` is set on a namespace and contains this rule name.
+	IgnoreNamespaceOverride bool `json:"ignoreNamespaceOverride"`
+
+	MatchIntegratedRegistry bool     `json:"matchIntegratedRegistry"`
+	MatchRegistries         []string `json:"matchRegistries"`
+
+	// AllowResolutionFailure allows the subsequent rules to be bypassed if the integrated registry does
+	// not have access to image metadata (no image exists matching the image digest).
+	AllowResolutionFailure bool `json:"allowResolutionFailure"`
+
+	MatchDockerImageLabels []ValueMatch     `json:"matchDockerImageLabels"`
```

Comment:
This only provides more power if we're pulling from a non-integrated registry, right?  Otherwise, its just a substitute for external marking rules.